### ISSUE TITLE
proposal: `avoid_unnecessary_decimal_trailing_zeros`

### ISSUE THREAD

**Issue by om-ha** [dart-lang/linter#3033]
### Details

Developers might be inconsistent between using a decimal trailing zeros and not using it when type is inferred/explicitly defined as `double`.

### Good Examples 

```dart
const goodSizedBox = SizedBox(height: 42);
double goodConsistentNumber = 42;
double goodPreciseNumber = 42.001
```

### Bad Examples

```dart
const badSizedBox = SizedBox(height: 42.0);
double badConsistentNumber = 42.00;
double badPreciseNumber = 42.001000
```

### Discussion

Honestly this doesn't bring any critical value, it just keeps codebases consistent.
